src/services/feedPage.services.ts

```diff
@@ -0,0 +1,64 @@
+import axios from "axios";
+import { retrieveFiles, storeFiles } from "../integrations/web3storage";
+import { CreatedWork, ICreatedWork } from "../models/createdWork.schema";
+import { User } from "../models/user.model"
+import { executeUserOp, getPartialUserOp } from "../integrations/biconomy";
+import { ethers } from "ethers";
+import { getNetworkConfig } from "../utils/network.utils";
+import { HttpException } from "../utils/exceptions.utils";
+import { BAD_REQUEST, ONE, ZERO } from "../utils/constants.utils";
+import { ICreateWorkSchema, ICreatorWorkMetadata } from "../models/interfaces.models";
+
+class FeedPageService {
+    getPaginatedWorks = async (page: number, count: number, nameOfWork: string, nameOfCreator: string, medium: string, licenseType: string, startDate: string, endDate: string) => {
```

Comment:
Remember there is no way the user will specify what they are sending to the back end.

Its a single search bar, so the user simply comes there to type a search query and that query could either be a creators name or name of a created work.

On our end, we should be checking

1. If we have we have a creator whose name includes the search query
2.  If we have we have any created work whose name includes the search query

Based on this we do not need to have `nameOfCreator` and `nameOfWork`, we should just have a `searchQuery` that will represent whatever the user has typed in.

-----

This is also lot of input parameters and it we could easily make a mistake while collecting. So declare a type and pass the type here instead

For example:
```typescript
type GetPaginatedWorksRequest = {
  page?: number,
  count?: number
  searchParam?: string,
  medium?: string,
  licenseType?: string,
  startDate?: string,
  endStart?: string
}
```

getPaginatedWorks = async (request: GetPaginatedWorksRequest){}

Reply:
Remember if I do not pass in any of these parameters, the endpoint should not fail

Instead it should return all items with a default pagination (eg 10)

So if I call this endpoint with no extra information, as long as there is data in the DB I should get the first ten items from the DB

Reply:
This is already satisfied as query is initialized to { } and find( { } ) returns all the items from DB with a default pagination

Reply:
Made changes to how search query is checked.
Declared type GetPaginatedWorksRequest.